File: Data Structures & Algorithms/anagram-groups/submission-1.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();

        for (String word : strs) {
            int[] w = new int[26];
            for (char c : word.toCharArray()) {
                w[c - 'a']++;
            }

            String key = Arrays.toString(w);
            map.putIfAbsent(key, new ArrayList<>());
            map.get(key).add(word);
        }

        List<List<String>> res = new ArrayList<>();
        map.forEach((k, v) -> {
            res.add(v);
        });
        return res;
    }
}

File: Data Structures & Algorithms/non-cyclical-number/submission-1.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n not in seen:
            seen.add(n)
            res = 0
            while n:
                res += (n%10) ** 2
                n //= 10
            if res == 1:
                return True
            n = res
        return False
